corehq/apps/app_manager/dbaccessors.py

```diff
@@ -235,6 +235,23 @@ def get_built_app_ids_with_submissions_for_app_id(domain, app_id, version=None):
     return [result['id'] for result in results]
 
 
+def get_built_app_ids_with_submissions_for_app_ids_and_versions(domain, app_ids_and_versions=None):
```

Comment:
I had to modify the way we get built app ids for cases. Before it would return some apps that didn't have submissions and wouldn't be processed. This now ensures we only fetch apps with submissions. This is important for the new Show Deleted because number of processed apps will always be greater than 0 in some cases since we don't process ones without submissions. And thus the modal will always pop up.

Another thing about this dbaccessor is that it makes view calls in a loop. Rather than write another view for just this function, I utilized the already established view. It goes through each app id in the domain and gets all build ids for that app. I don't expect this loop to ever be extremely long, but if you have lots of apps in your domain it can make a lot of view calls.